airflow/providers/openlineage/extractors/manager.py

```diff
@@ -0,0 +1,150 @@
+
+        # TODO: use airflow config with OL backup
+        env_extractors = os.getenv("OPENLINEAGE_EXTRACTORS")
+        if env_extractors is not None:
+            for extractor in env_extractors.split(";"):
+                extractor: type[BaseExtractor] = import_from_string(extractor.strip())
+                for operator_class in extractor.get_operator_classnames():
+                    self.extractors[operator_class] = extractor
+
+    def add_extractor(self, operator: str, extractor: type[BaseExtractor]):
+        self.extractors[operator] = extractor
+
+    def extract_metadata(self, dagrun, task, complete: bool = False, task_instance=None) -> OperatorLineage:
+        extractor = self._get_extractor(task)
+        task_info = (
+            f"task_type={task.task_type} "
+            f"airflow_dag_id={task.dag_id} "
+            f"task_id={task.task_id} "
+            f"airflow_run_id={dagrun.run_id} "
+        )
+
+        if extractor:
+            # Extracting advanced metadata is only possible when extractor for particular operator
+            # is defined. Without it, we can't extract any input or output data.
+            try:
+                self.log.debug("Using extractor %s %s", extractor.__class__.__name__, str(task_info))
+                if complete:
+                    task_metadata = extractor.extract_on_complete(task_instance)
+                else:
+                    task_metadata = extractor.extract()
+
+                self.log.debug("Found task metadata for operation %s: %s", task.task_id, str(task_metadata))
+                if task_metadata:
+                    if (not task_metadata.inputs) and (not task_metadata.outputs):
+                        inlets = task.get_inlet_defs()
+                        outlets = task.get_outlet_defs()
+                        self.extract_inlets_and_outlets(task_metadata, inlets, outlets)
+
+                    return task_metadata
+
+            except Exception as e:
+                self.log.exception(
+                    "Failed to extract metadata using found extractor %s - %s %s", extractor, e, task_info
+                )
+        else:
+            self.log.debug("Unable to find an extractor %s", task_info)
+
+            # Only include the unkonwnSourceAttribute facet if there is no extractor
+            task_metadata = OperatorLineage(
+                run_facets={
+                    "unknownSourceAttribute": UnknownOperatorAttributeRunFacet(
+                        unknownItems=[
+                            UnknownOperatorInstance(
+                                name=get_operator_class(task).__name__,
+                                properties={attr: value for attr, value in task.__dict__.items()},
+                            )
+                        ]
+                    )
+                },
+            )
+            inlets = task.get_inlet_defs()
+            outlets = task.get_outlet_defs()
```

Comment:
These are deprecated now
```suggestion
            inlets = task.inlets
            outlets = task.outlets
```

(Added/changed in #25767, which was 2.4. So maybe this is fine as it its?

Reply:
Fixed.